core/Integer.Format.savi

```diff
@@ -17,110 +17,27 @@
 :struct val Integer.Format.Decimal(T Integer(T)'val)
   :is IntoString
 
-  :let _value T
-  :new val _new(@_value)
+  :let _bcd U64.BCD
+  :let _is_negative Bool
 
-  :fun into_string(out String'iso) String'iso
-    value = @_value
-
-    // TODO: Avoid this hacky workaround for lack of numeric literals here.
+  :new val _new(value T)
     zero = value.zero
-    one = value.one
-    ten = one + one + one + one + one + one + one + one + one + one
-
-    // Fast path for zero - it always has exactly one digit.
-    if (value == zero) (
-      out.push_byte('0')
-      return --out
-    )
-
-    // If the value is less than zero, we need to negate it to make it positive
-    // before we try to append its representation into the given string,
-    // and we'll append a negative sign byte first.
     if (value < zero) (
       value = zero - value
-      out.push_byte('-')
+      @_is_negative = True
+    |
+      @_is_negative = False
     )
+    @_bcd = U64.BCD.new(value.u64)
 
-    // Convert the value to a U64 so we don't have to specialize code by size.
-    // TODO: It would likely be slightly faster to specialize by size here.
-    u64 = value.u64
-
-    // Get the number of digits to print and the initial divisor to use.
-    digit_count = @_overestimated_digit_count(value.significant_bits)
-    divisor = try (@_powers_of_10[(digit_count - 1).usize]! | 10)
-
-    // For some numbers, our initial estimate is one higher than the correct
-    // digit count, so in such cases we adjust our values down by one digit.
-    if (u64 < divisor) (
-      digit_count -= 1
-      divisor = divisor / 10
-    )
-
-    // Print the digits from left to right (most to least significant).
-    // This works because the divisor is decreasing by a factor of 10 each time.
-    digit_count.times -> (
-      out.push_byte((u64 / divisor % 10).u8 + '0')
-      divisor = divisor / 10
+  :fun into_string(out String'iso) String'iso
+    if @_is_negative (
+      out.push_byte('-')
     )
-
-    --out
+    --@_bcd.into_string(--out)
```

Comment:
Consume/`--` only works on local variables. There should be a nice compiler error for this but right now there is not - it generates an internal error in one of the unused passes.

```suggestion
    @_bcd.into_string(--out)
```